src/main/java/za/ac/sun/cs/search/adversarialagent/Board.java

```diff
@@ -0,0 +1,164 @@
+package za.ac.sun.cs.search.adversarialagent;
+
+import java.util.ArrayList;
+import java.lang.StringBuilder;
+
+import za.ac.sun.cs.search.adversarialagent.Move;
+
+public class Board {
+    private int m;
+    private int n;
+    private int k;
+    private int[][] board;
+    private int player;
+
+    public Board (int m, int n, int k) {
+        this.m = m;
+        this.n = n;
+        this.k = k;
+        this.board = new int[m][n];
+        this.player = 1;
+    }
+
+    public ArrayList<Move> getLegalMoves (int player) {
```

Comment:
Why does this need to know who the player is?

Reply:
This being `getLegalMoves` :hamster:

Reply:
It was a formality, we can remove it.

Reply:
Yeah I think we can strip it for the time being. If need be, we'll extend and override later :+1: